src/wallet/migrate.cpp

```diff
@@ -16,7 +17,25 @@ std::unique_ptr<DatabaseBatch> BerkeleyRODatabase::MakeBatch(bool flush_on_close
 
 bool BerkeleyRODatabase::Backup(const std::string& dest) const
 {
-    return false;
+    fs::path src(m_filepath);
+    fs::path dst(fs::PathFromString(dest));
+
+    if (fs::is_directory(dst)) {
+        dst = BDBDataFile(dst);
+    }
+    try {
+        if (fs::exists(dst) && fs::equivalent(src, dst)) {
+            LogPrintf("cannot backup to wallet source file %s\n", fs::PathToString(dst));
+            return false;
+        }
+
+        fs::copy_file(src, dst, fs::copy_options::overwrite_existing);
+        LogPrintf("copied %s to %s\n", fs::PathToString(m_filepath), fs::PathToString(dst));
+        return true;
+    } catch (const fs::filesystem_error& e) {
+        LogPrintf("error copying %s to %s - %s\n", fs::PathToString(m_filepath), fs::PathToString(dst), fsbridge::get_filesystem_error_message(e));
+        return false;
+    }
```

Comment:
> Is this used in this pull request?

No, it's used in the followup.

> I wonder if it could make sense to return the error string to the caller, assuming this is called by RPC eventually?

`Backup` is a part of the `WalletDatabase` parent class, so changing this would also mean changing the `Backup` of existing database classes, and all of their callers. I think that's something that would be worth doing in a followup, but not this PR.